pkg/internal/apply/layerApplier.go

```diff
@@ -219,13 +275,35 @@ func (a KubectlLayerApplier) Apply(layer layers.Layer) (err error) {
 		return err
 	}
 
-	err = a.logAddons(hrs, layer)
-
 	// TODO: Add an ownerRef to each deployed HelmRelease the points back to this AddonsLayer (needed for Prune)
+	err = a.addOwnerRefs(layer, hrs)
+	if err != nil {
+		return err
+	}
+
+	err = a.applyHelmReleases(layer, hrs)
+	if err != nil {
+		return err
+	}
+
+	err = a.logAddons(hrs, layer)
+	if err != nil {
+		return err
+	}
 
 	// TODO: Watch all HelmRelease resources applied for this AddonsLayer until all are success or fail or timeout
 
-	return err
+	return nil
+}
+
+// ApplyIsRequired returns true if any resources need to be applied for this AddonsLayer
+func (a KubectlLayerApplier) ApplyIsRequired(layer layers.Layer) (applyIsRequired bool, err error) {
+	return true, nil
```

Comment:
Guessing these functions are going to have code added in the future.

Reply:
Nope.